lib/ansible/modules/storage/netapp/na_ontap_snapshot.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/python
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: na_ontap_snapshot
+short_description: Manage NetApp Sanpshots
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author:
+- Chris Archibald (carchi@netapp.com), Kevin Hutton (khutton@netapp.com)
+description:
+- Create/Modify/Delete Ontap snapshots
+options:
+  state:
+    description:
+    - If you want to create/modify a snapshot, or delete it.
+    choices: ['present', 'absent']
+    default: present
+  snapshot:
+    description:
+    - Name of the snapshot to be created.
+    - The maximum string length is 256 characters.
+    required: true
+  volume:
+    description:
+    - Name of the volume on which the snapshot is to be created.
+    required: true
+  async_bool:
+    description:
+    - If true, the snapshot is to be created asynchronously.
+    type: bool
+  comment:
+    description:
+    - A human readable comment attached with the snapshot.
+    - The size of the comment can be at most 255 characters.
+  snapmirror_label:
+    description:
+    - A human readable SnapMirror Label attached with the snapshot.
+    - Size of the label can be at most 31 characters.
+    required: false
+  ignore_owners:
```

Comment:
maybe `ignore_owners_on_removal` or something (to be clear it's only applicable during deletion)?